### PR TITLE
Created a new column to track training categories that are not requir…

### PR DESCRIPTION
**Pull request by Ohsudev**
…ed as training requirements. This set is for the EHR Module version of additional code.

#### Changes
Created a new column to track training categories that are not required as training requirements. This set is for the EHR Module version.  The other set of codes that complements this feature branch resides on "onprcEHRModules"/

### CODE REVIEW

**Review by kollil — APPROVED** [review]
Raymond, The changes look good.

**Review by kollil — COMMENTED** [review]
Raymond, I forgot to mention if the new db field is added to the ehr_compliancedb.xml file?

File: EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/EHR_ComplianceDB-12.40-12.41.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE ehr_compliancedb.RequirementsPerCategory add TrackingFlag varchar(100);
```
Comment (labkey-jeckels) [inline EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/EHR_ComplianceDB-12.40-12.41.sql]:
```suggestion
ALTER TABLE ehr_compliancedb.RequirementsPerCategory add TrackingFlag nvarchar(100);
```

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
As Lakshmi commented, the new column should get an entry in ehr_compliancedb.xml

**Comment by Ohsudev** [comment]
HI Josh.  Do I still need to make an entry into  “ehr_compliancedb.xml” despite having the following code already include into “onprcEHRModules” repository part of my feature branch?

C:\Test_20.11\server\modules\onprcEHRModules\ONPRC_EHR_ComplianceDB\resources\queries\EHR_ComplianceDB\RequirementsPerCategory.query.xml
    <column columnName="TrackingFlag">
                <isUserEditable>true</isUserEditable>
                <nullable>true</nullable>
                <isHidden>false</isHidden>
            </column>

Thanks,
Raymond

From: Josh Eckels ***@***.***>
Sent: Wednesday, May 26, 2021 10:43 AM
To: LabKey/ehrModules ***@***.***>
Cc: Raymond Blasa ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [LabKey/ehrModules] Created a new column to track training categories that are not requir… (#186)


@labkey-jeckels requested changes on this pull request.

As Lakshmi commented, the new column should get an entry in ehr_compliancedb.xml

________________________________

In EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/EHR_ComplianceDB-12.40-12.41.sql<https://github.com/LabKey/ehrModules/pull/186#discussion_r639983816>:

> @@ -0,0 +1 @@



+ALTER TABLE ehr_compliancedb.RequirementsPerCategory add TrackingFlag varchar(100);


⬇️ Suggested change

-ALTER TABLE ehr_compliancedb.RequirementsPerCategory add TrackingFlag varchar(100);



+ALTER TABLE ehr_compliancedb.RequirementsPerCategory add TrackingFlag nvarchar(100);

**Comment by labkey-jeckels** [comment]
@Ohsudev yes, please. We have automated tests that compare the list of columns in the centralized schema XML file (ehr_compliancedb.xml in this case) against what's actually in the database. That helps us make sure that the Postgres and SQLServer versions of the schemas don't diverge for modules that support both DBs.

**Comment by Ohsudev** [comment]
Added a new file as per Josh's and Lakshmi's  request to be included into this feature branch.  Please review again for me.

File: EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/EHR_ComplianceDB-12.40-12.41.sql (hunk 1 above)
Comment (labkey-jeckels) [reply EHR_ComplianceDB/resources/schemas/dbscripts/sqlserver/EHR_ComplianceDB-12.40-12.41.sql]: I'm going to go ahead and commit this suggested change.

**Review by labkey-jeckels — APPROVED** [review]

**Comment by Ohsudev** [comment]
No longer needed.

**Pull request closed without merging**